pennylane/labs/resource_estimation/ops/qubit/parametric_ops_multi_qubit.py

```diff
@@ -47,6 +49,39 @@ def resource_params(self):
     def resource_rep(cls, num_wires):
         return re.CompressedResourceOp(cls, {"num_wires": num_wires})
 
+    @classmethod
+    def adjoint_resource_decomp(cls) -> Dict[re.CompressedResourceOp, int]:
+        return {cls.resource_rep(): 1}
+
+    @staticmethod
+    def controlled_resource_decomp(
+        num_ctrl_wires,
+        num_ctrl_values,
+        num_work_wires,
+        num_wires,
+    ) -> Dict[re.CompressedResourceOp, int]:
+        if num_ctrl_values == 0:
+            cnot = re.ResourceCNOT.resource_rep()
+            ctrl_rz = re.ResourceControlled.resource_rep(
+                base_class=re.ResourceRZ,
+                base_params={},
+                num_ctrl_wires=num_ctrl_wires,
+                num_ctrl_values=num_ctrl_values,
+                num_work_wires=num_work_wires,
+            )
+
+            gate_types = {}
+            gate_types[cnot] = 2 * (num_wires - 1)
+            gate_types[ctrl_rz] = 1
+
+            return gate_types
```

Comment:
Can we not handle `num_ctrl_values > 0` by adding X gates?

Reply:
We can! However if you look at the logic in `re.ResourceControlled` it first tries to expand the op, if it encounters an error, it adds the Xs and then tries the controlled method again. So here I am relying on that logic to add the Xs